cpp/splinepy/py/py_multi_patch.hpp

```diff
@@ -0,0 +1,2412 @@
+          field_mismatch_info += "[mismatch error from the field with index ("
+                                 + std::to_string(j) + ")]\n" + e.what();
+        }
+      }
+    };
+
+    // raise, if there were error.
+    if (field_mismatch_info.size() != 0) {
+      splinepy::utils::PrintAndThrowError(field_mismatch_info);
+    }
+
+    // all good, extend to list
+    field_list_ += fields;
+  }
+
+  py::list GetFields() { return field_list_; }
+};
+
+inline void add_multi_patch(py::module& m) {
+
+  // returns [connectivity, vertex_ids, edge_information, boundaries]
+  m.def("retrieve_mfem_information",
+        &splinepy::py::RetrieveMfemInformation,
+        py::arg("corner_vertices"),
+        py::arg("tolerance"));
+  m.def("interfaces_from_boundary_centers",
+        &splinepy::py::InterfacesFromBoundaryCenters,
+        py::arg("face_center_vertices"),
+        py::arg("tolerance"),
+        py::arg("para_dim"));
+  m.def("extract_all_boundary_splines",
+        &splinepy::py::ExtractAllBoundarySplines,
+        py::arg("splines"),
+        py::arg("interfaces"),
+        py::arg("nthreads") = 1);
+  m.def("orientations",
+        &splinepy::py::GetBoundaryOrientations,
+        py::arg("splines"),
+        py::arg("base_ids"),
+        py::arg("base_face_ids"),
+        py::arg("neighbor_ids"),
+        py::arg("neighbor_face_ids"),
+        py::arg("tolerance"),
+        py::arg("nthreads") = 1);
+  m.def("boundaries_from_continuity",
+        &splinepy::py::AddBoundariesFromContinuity,
+        py::arg("boundary_splines"),
+        py::arg("boundary_interfaces"),
+        py::arg("global_interfaces"),
+        py::arg("tolerance"),
+        py::arg("nthreads") = 1);
+
+  py::class_<splinepy::py::PyMultiPatch,
+             std::shared_ptr<splinepy::py::PyMultiPatch>>
+      klasse(m, "PyMultiPatch");
+
+  klasse.def(py::init<>())
+      .def(py::init<py::list&, const int, const bool>())
+      .def_readwrite("n_default_threads", &PyMultiPatch::n_default_threads_)
+      .def_readwrite("same_parametric_bounds",
+                     &PyMultiPatch::same_parametric_bounds_)
+      .def_readwrite("tolerance", &PyMultiPatch::tolerance_)
+      .def("clear", &PyMultiPatch::Clear)
+      .def_property_readonly("para_dim", &PyMultiPatch::ParaDim)
+      .def_property_readonly("dim", &PyMultiPatch::Dim)
+      .def_property_readonly("name", &PyMultiPatch::Name)
+      .def_property_readonly("whatami", &PyMultiPatch::WhatAmI)
+      .def_property("patches",
+                    &PyMultiPatch::GetPatches,
+                    &PyMultiPatch::SetPatchesDefault)
+      .def("sub_multi_patch", &PyMultiPatch::SubMultiPatch)
+      .def("sub_patch_centers", &PyMultiPatch::SubPatchCenters)
+      .def("interfaces", &PyMultiPatch::Interfaces)
+      .def("boundary_patch_ids", &PyMultiPatch::BoundaryPatchIds)
+      .def("boundary_multi_patch", &PyMultiPatch::BoundaryMultiPatch)
+      .def("evaluate",
```

Comment:
as mentioned in #141 the goal is to provide as many single-spline functionality in "multiple-spline" scenarios. Since we abandoned `SplineList` idea, MultiPatch should take over